compiler/passes/insertWideReferences.cpp

```diff
@@ -832,6 +832,9 @@ static void propagateVar(Symbol* sym) {
           debug(sym, "_val of ref %s (%d) needs to be wide\n", lhs->cname, lhs->id);
           setValWide(lhs);
         }
+        else if (lhs->type == dtString) {
```

Comment:
Can you provide some reasoning in a comment for why strings must always be widened (specally) in this way?  Which test case was failing? and why does this seem to be the right fix?